### PR TITLE
SEAT-232 Now handling small touch events as map selections

### PR DESCRIPTION
**Pull request by mayavera**
On Android browsers, very small movements during touch events (after the user touches the screen and before they let go) were being interpreted as requests to move the map. I've implemented a sort of [_noise gate_](https://en.wikipedia.org/wiki/Noise_gate). In short, it treats very small touch motions as selections. This gate is configurable via the `MAX_SELECT_TRANSLATION_DISTANCE` constant in `src/TicketMap/index.tsx`.

I also removed the existing drag detection logic.

### CODE REVIEW

File: src/TicketMap/index.tsx [hunk 1]
```diff
@@ -368,13 +370,38 @@ export default class TicketMap extends Component<Props & DefaultProps, State> {
 
   onClick = () => this.doSelect()
 
-  onTouchMove = () => this.setState({ isDragging: true })
+  onTouchStart = (event: React.TouchEvent<HTMLElement>) => {
+    if (event.changedTouches.length !== 1) {
+      return
+    }
+
+    const [ touchID ] = Object.getOwnPropertyNames(event.changedTouches)
```
Comment (Pharserror) [inline src/TicketMap/index.tsx]: If you return unless there's only ever 1 `changesTouches` then can't we always assume touchId to be `"0"`?

Comment (mayavera) [reply src/TicketMap/index.tsx]: I suppose so. I just read up a bit more on the `TouchList` interface and it seems like you can get touches by index using `.item()`. Man, this is a strange interface. I'm wondering why `changedTouches` isn't simply an array...

File: src/TicketMap/index.tsx [hunk 2]
```diff
@@ -368,13 +370,38 @@ export default class TicketMap extends Component<Props & DefaultProps, State> {
...
+    const touch = event.changedTouches[touchID as any]
+    const touchStart = this.state.touchStarts[touchID]
 
-  onTouchEnd = ({ target }: React.TouchEvent<HTMLElement>) => {
-    if (!this.state.isDragging) {
-      this.doSelect(this.getSectionFromTarget(target as HTMLElement))
+    const translationDistance = Math.sqrt(Math.pow(touchStart.x - touch.pageX, 2) + Math.pow(touchStart.y - touch.pageY, 2))
```
Comment (Pharserror) [inline src/TicketMap/index.tsx]: alright 👍 for🔺's 😎

Comment (Pharserror) [inline src/TicketMap/index.tsx]: woop woop math

Comment (mayavera) [reply src/TicketMap/index.tsx]: lol it's just a 2-d distance formula

File: src/TicketMap/index.tsx [hunk 3]
```diff
@@ -368,13 +370,38 @@ export default class TicketMap extends Component<Props & DefaultProps, State> {
...
+    if (translationDistance <= MAX_SELECT_TRANSLATION_DISTANCE) {
```
Comment (Pharserror) [inline src/TicketMap/index.tsx]: you did manage to get a phone and test that this distance didn't cause any unexpected behavior?

Comment (mayavera) [reply src/TicketMap/index.tsx]: yep

File: src/TicketMap/index.tsx (hunk 2 above)
Comment (Pharserror) [reply src/TicketMap/index.tsx]: let's hear it for 🔺's 👏

**Pull request merged**